modules/olm-updating-operator-catalog-image.adoc

```diff
@@ -0,0 +1,131 @@
+catalog images, administrators can keep their {product-title} cluster up to date
+with the latest Operators by capturing updates made to Red Hat’s App Registry
+catalogs. This is done by building and pushing a new Operator catalog image,
+then replacing the existing  CatalogSource’s `spec.image` parameter with the new
+image digest.
+
+For this example, the procedure assumes a custom `redhat-operators` catalog
+image is already configured for use with OperatorHub.
+
+.Prerequisites
+
+* A Linux workstation with unrestricted network access
+ifeval::["{context}" == "olm-restricted-networks"]
+footnoteref:[BZ1771329]
+endif::[]
+* `oc` version 4.3.5+
+* `podman` version 1.5.1+
+* Access to mirror registry that supports
+link:https://docs.docker.com/registry/spec/manifest-v2-2/[Docker v2-2]
+* OperatorHub configured to use custom catalog images
+
+.Procedure
+
+. On the workstation with unrestricted network access, authenticate with the
+target mirror registry:
++
+----
+$ podman login <registry_host_name>
+----
+
+. Build a new catalog image based on the `redhat-operators` catalog from
+link:https://quay.io/[quay.io], tagging and pushing it to your mirror registry:
++
+----
+$ oc adm catalog build \
+    --appregistry-org redhat-operators \//<1>
+    --to=<registry_host_name>:<port>/olm/redhat-operators:v2 \//<2>
+    [-a <path_to_registry_credentials>] \//<3>
+    [--insecure] <4>
+
+INFO[0013] loading Bundles                               dir=/var/folders/st/9cskxqs53ll3wdn434vw4cd80000gn/T/300666084/manifests-829192605
+...
+Pushed sha256:f73d42950021f9240389f99ddc5b0c7f1b533c054ba344654ff1edaf6bf827e3 to example_registry:5000/olm/redhat-operators:v2
+----
+<1> Organization (namespace) to pull from an App Registry instance.
+<2> Name your catalog image and include a tag, for example, `v2` because it is the updated catalog.
+<3> If required, specify the location of your registry credentials file for `podman` or `docker`.
+<4> If you do not want to configure trust for the target registry, add the `--insecure` flag.
+
+. Mirror the contents of your catalog to your target registry. The following
+`oc adm catalog mirror` command extracts the contents of your custom Operator
+catalog image to generate the manifests required for mirroring and mirrors the
+images to your registry:
++
+----
+$ oc adm catalog mirror \
+    <registry_host_name>:<port>/olm/redhat-operators:v2 \//<1>
+    <registry_host_name>:<port> \
+    [-a <path_to_registry_credentials>] \//<2>
+    [--insecure] <3>
+
+mirroring ...
+----
+<1> Specify your new Operator catalog image.
+<2> If required, specify the location of your registry credentials file for `podman` or `docker`.
+<3> If you do not want to configure trust for the target registry, add the `--insecure` flag.
+
+. Apply the newly-generated manifests:
++
+----
+$ oc apply -f ./redhat-operators-manifests
+----
++
+[IMPORTANT]
+====
+If any invalid source tags are found, remove the offending mappings from your
```

Comment:
We should not need this statement as we should move past those now @ecordell correct me if i'm wrong?

Reply:
:+1: Wasn't sure if that had shipped or not, but looks like it did with 4.3.3 via https://bugzilla.redhat.com/show_bug.cgi?id=1797730.

Reply:
Note that I bumped the `oc` version mentioned in prerequisites for this procedure to 4.3.5 to take advantage of the latest changes/fixes. I'll do the same in the other catalog procedures in this topic too.